### DISCUSSION TITLE
'CI-long-timeout' label

### DISCUSSION THREAD

**Discussion by ilya-lavrenov** [Homebrew/discussions#4075] [Everyday usage]
### Output of `brew config`

```shell
Not applicable
```

### Output of `brew doctor`

```shell
Not applicable
```

### Description of issue

Hi brew maintainers,

I have several PRs on review and one of them is waiting for almost a month. 
I fully understand status of this project and I don't want to complain, but rather want to thank you for all your effort and want to provide a suggestion.

I don't know a lot of details, but what I see that sometimes CI fails because of time-out and maintainers have to set different labels to make timeout longer. E.g. to denote long build, you set `long build`, to denote `CI-long-timeout` that this specific PR can use longer timeout. And in my PRs I see:
```
Error: PR requires the CI-long-timeout label but it is not set!
[142](https://github.com/Homebrew/homebrew-core/actions/runs/3797197200/jobs/6457961782#step:2:143)
Error: If the longer timeout is not required, remove the "long build" label.
[143](https://github.com/Homebrew/homebrew-core/actions/runs/3797197200/jobs/6457961782#step:2:144)
Error: Otherwise, add the "CI-long-timeout" label.
[144](https://github.com/Homebrew/homebrew-core/actions/runs/3797197200/jobs/6457961782#step:2:145)
Error: No more than 2 PRs at a time may use "CI-long-timeout"
```

Looks like the process of adding such labels is manual and it make the process of handling such long build to be unoptimal.
Do you think it can be automated with GitHub actions?

The workflow:
- when PR is originally submitted and failed by timeout, `long build` is automatically set.
- create new workflow which automatically runs every XX minutes to check whether some of the PR holds `CI-long-timeout`. If no, it takes one PR from the list of labeled with `long build` and assigns label `CI-long-timeout` and reruns checks. Once checks on such PR have been finished, both labels are automatically removed.

Please, let me know if I correctly understand your workflow and whether it can be helpful.

Thanks.

**Comment by SMillerDev** [comment]
The first part is useful. The second part would likely be difficult to make, but if you can manage I think it would be useful to automate one of the two queue slots.
